optuna/samplers/_cmaes.py

```diff
@@ -736,6 +736,9 @@ def _get_trials(self, study: "optuna.Study") -> List[FrozenTrial]:
                 _, value = max(t.intermediate_values.items())
                 if value is None:
                     continue
+
+                assert isinstance(value, float)
```

Comment:
If the multi-dimensional intermediate values are given, this line causes an AssertionError. This means the sampler implicitly assumes that intermediate values are one-dimensional. For instance, the pruner may issue a warning in such cases, but is it okay for the sampler not to explicitly produce error messages or warnings?

Reply:
This comment can be applied to other samplers.